waiter/src/waiter/core.clj

```diff
@@ -794,24 +798,31 @@
                                       (fn service-description->service-id [service-description]
                                         (sd/service-description->service-id service-id-prefix service-description)))
    :service-id->idle-timeout (pc/fnk [[:settings [:token-config token-defaults]]
-                                      service-id->service-description-fn token->token-hash token->token-metadata]
+                                      service-id->service-description-fn service-id->source-tokens-entries-fn
+                                      token->token-hash token->token-metadata]
                                (fn service-id->idle-timeout [service-id]
                                  (sd/service-id->idle-timeout
-                                   service-id->service-description-fn token->token-hash token->token-metadata
-                                   token-defaults service-id)))
+                                   service-id->service-description-fn service-id->source-tokens-entries-fn
+                                   token->token-hash token->token-metadata token-defaults service-id)))
    :service-id->password-fn (pc/fnk [[:scheduler service-id->password-fn*]]
                               service-id->password-fn*)
    :service-id->service-description-fn (pc/fnk [[:scheduler service-id->service-description-fn*]]
                                          service-id->service-description-fn*)
+   :service-id->source-tokens-entries-fn (pc/fnk [[:curator kv-store]]
+                                           (partial sd/service-id->source-tokens-entries kv-store))
    :start-new-service-fn (pc/fnk [[:scheduler scheduler]
                                   [:state authenticator start-app-cache-atom task-threadpool]
-                                  store-service-description-fn]
-                           (fn start-new-service [{:keys [service-id] :as descriptor}]
+                                  store-service-description-fn store-source-tokens-fn]
+                           (fn start-new-service [{:keys [service-id source-tokens] :as descriptor}]
                              (let [run-as-user (get-in descriptor [:service-description "run-as-user"])]
                                (auth/check-user authenticator run-as-user service-id))
-                             (service/start-new-service
-                               scheduler descriptor start-app-cache-atom task-threadpool
-                               :pre-start-fn #(store-service-description-fn descriptor))))
+                             (let [pre-start-fn (fn pre-start-new-service-fn []
```

Comment:
Won't `request->descriptor` be called on this path?  Is a change in this fn necessary?